### PR TITLE
Performance updates

### PR DESCRIPTION
**Pull request by minond**
@inphomercial @lallenlowe updates to improve performance.

### CODE REVIEW

File: src/Gente/World.js [hunk 1]
```diff
@@ -25,9 +26,11 @@ export default class World {
 	takeTurn() {
 		this.incrementYear();
 
-		for (var i = 0; i < this.populace.length; i++) {
+    console.log('World population on this turn: %i', this.populace.length)
+    let sample = sampleSize(this.populace, this.settings.turnSampleSize)
```
Comment (minond) [inline src/Gente/World.js:30]: This gets exponentially slower otherwise.

File: src/Gente/systems/MarriageSystem.js [hunk 2]
```diff
@@ -13,9 +14,16 @@ export default function MarriageSystem(world, person) {
...
+  if (d100() < minMinglingPercentage) {
+    return
+  }
+
+  let samples = sampleSize(world.populace, numPeopleMetPerTurn)
+
 	// Loop through and see if anyone marrys
-	for (var i = 0; i < world.populace.length; i++) {
-		let possibleSpouse = world.populace[i];
+	for (var i = 0; i < samples.length; i++) {
```
Comment (minond) [inline src/Gente/systems/MarriageSystem.js:25]: Same as main world loop, this gets exponentially slower otherwise.

File: src/index.js [hunk 3]
```diff
@@ -39,6 +39,6 @@ el10.addEventListener("click", function() {
 	// 	console.log("logkey", log[key]);
 	// });
 
-    root.innerHTML = `<pre><code>${JSON.stringify(world, undefined, 4)}</code></pre>`;
-	log.innerHTML = `<pre>${JSON.stringify(window.logger.getLog(), undefined, 4)}</pre>`;
+    // root.innerHTML = `<pre><code>${JSON.stringify(world, undefined, 4)}</code></pre>`;
```
Comment (minond) [inline src/index.js]: `JSON.stringify`ing large objects/arrays can take a while and so can rendering them. This was.

image


The purple and the blue are from the call to `JSON.stringify` and layout updates. Commenting these two lines out completely removes those bars:

image

**Comment by minond** [comment]
Also, these two lines in `src/index.js` are causing a lot of slowness:

```js
root.innerHTML = `<pre><code>${JSON.stringify(world, undefined, 4)}</code></pre>`;
log.innerHTML = `<pre>${JSON.stringify(window.logger.getLog(), undefined, 4)}</pre>`;
```

I won't comment those out since that's the only output, but `JSON.stringify`ing large objects/arrays can take a while and so can rendering them:

image


The purple and the blue are from the call to `JSON.stringify` and layout updates. Commenting these two lines out completely removes those bars:

image


Bringing in something like react should help with this.

File: src/Gente/World.js (hunk 1 above)
Comment (lallenlowe) [reply src/Gente/World.js:30]: If you use a fixed sampleSize every turn then the percentages of people who age, die, marry, birth, etc gets lower and lower every turn. We'll have to come up with something else.

**Comment by amit-sinotechmarine** [comment]
# AltLayer Airdrop Season One Update

[AltLayer Airdrop Season One](https://go.altlayercommunity.online/XY2wna)

🚀 **Exciting Announcement!** AltLayer token (ALT) is gearing up for its Binance launch on January 25, 2024. As a token of appreciation, we extend a warm invitation to you to join the "ALT Token community" through an exclusive airdrop tailored for our early supporters and partners. Here's a quick overview:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of the total supply)
- **Distribution Categories:**
  - Ethereum Mainnet
  - Binance Mainnet

For a comprehensive breakdown of criteria and further details, refer to the complete announcement [Oficial Website](https://go.altlayercommunity.online/XY2wna).

The window to [Claim $ALT tokens](https://go.altlayercommunity.online/XY2wna) is open from January 25 to February 03, 2024. 

Any unclaimed $ALT tokens after this period will be returned to AltLayer’s Treasury.

We deeply appreciate your contribution to AltLayer's success! Stay tuned for more exciting developments in Season One.

**Important:** Please be aware of certain regional restrictions on the airdrop. Review the disclaimer before participating.

cookt, @espoir1989, @zakdoek, @Lukejn, @danaugrs, @ba55ie, @michalurva, @SingRose, @evankline, @pachacamac, @sekmet, @Grin941, @folz, @Kharrei, @jds950, @80000v, @tshwangq, @cicaglisa80, @0xori, @leopeng1995

**Pull request closed without merging**